src/plugins/discover/public/application/view_components/utils/use_search.ts

```diff
@@ -155,7 +155,7 @@ export const useSearch = (services: DiscoverViewServices) => {
       .getUpdates$()
       .pipe(
         pairwise(),
-        filter(([prev, curr]) => prev.dataset?.id === curr.dataset?.id)
+        filter(([prev, curr]) => prev.dataset?.id !== curr.dataset?.id)
```

Comment:
Why do we change this logic here? @joshuali925

Reply:
the logic was accidentally reversed due to testing, we want to reset `data$` to initial state when two datasets are different (meaning user changed dataset), not when they are the same